End-to-End-Tests.md

```diff
@@ -115,12 +115,17 @@ The challenge in writing robust E2E tests is making sure to always include a wai
 
 ### If the end-to-end tests are failing on your PR
 
+Most E2E flakes are a result of race conditions between the E2E test code, the simulated user, and the server code being tested. The speeds of these three processes, and therefore whether a given race condition causes flakes at an appreciable rate, can vary depending on the execution environment. For example, some race conditions might not yield a flake on faster computers because there the "correct" process always wins the race.
+We focus almost exclusively on fixing flakes that appear on CI, so flakes may appear more often in other execution environments (e.g. on your local machine) that differ substantially from the CI runners. We usually debug E2E tests on CI so that our debugging environment is as close as possible to the real thing. We would imagine that our CI runners are pretty fast (since they don't have to run a GUI, for example), so we would expect more flakes on slower computers. That's just based on intuition and my own experience though.
```

Comment:
Please add a blank line between paragraphs, otherwise github doesn't show a break. Also, let's remove "and my own experience" since this is a general wiki page, not one I alone wrote

Reply:
Done